File: android/app/src/main/java/app/releaf/mobile/ui/components/ActivityTimeline.kt
/*
 * ActivityTimeline.kt
 *
 * Bramble-garland activity timeline. A serpentine vine runs down the
 * left of a cream card. Each entry is a 5-petal flower in one of the
 * four leaf themes; between entries the vine carries a stout leaf and
 * a berry trio sitting on the vine line.
 *
 * Source spec: design-system/timeline-vine-bramble-garland.html.
 *
 * Layout is fixed-height: each entry occupies `EntryStrideDp` (90dp)
 * so the vine bezier can be plotted from known y positions. If a
 * dynamic-height variant is needed later, swap the Canvas overlay for
 * a custom Layout that measures children and feeds y positions back.
 */

package app.releaf.mobile.ui.components

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.DrawScope
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.graphics.drawscope.rotate
import androidx.compose.ui.graphics.drawscope.translate
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.releaf.mobile.ui.theme.AccentPalettes
import app.releaf.mobile.ui.theme.AccentPaletteId
import app.releaf.mobile.ui.theme.AppAccent
import app.releaf.mobile.ui.theme.AppColors
import app.releaf.mobile.ui.theme.AppSpacing
import app.releaf.mobile.ui.theme.AppTypography
import java.util.UUID

// region Public model

enum class ActivityProminence {
    /** Featured entry — bigger flower (today, captured highlights). */
    Featured,

    /** Routine entry — smaller flower. */
    Routine,
}

data class ActivityEntry(
    val date: String,
    val title: String,
    val preview: String? = null,
    val theme: AccentPaletteId,
    val prominence: ActivityProminence = ActivityProminence.Routine,
    val id: String = UUID.randomUUID().toString(),
)

// endregion

// region Layout constants — match the SwiftUI mirror so the two stay diff-able.

/** Width of the leading column reserved for the vine + flower marker. */
private val LeadColumnDp: Dp = 64.dp

/** Vertical distance between entry centers. */
private val EntryStrideDp: Dp = 64.dp

/** Where the vine sits inside the leading column (x within the card). */
private val VineXDp: Dp = 46.dp

/** Half-width of each vine bulge (control-x offset from VineXDp). */
private val VineBulgeDp: Dp = 20.dp

private val VineStrokeDp: Dp = 1.8.dp

/** Visible flower marker size — used to offset it onto the vine line. */
private val MarkerSizeDp: Dp = 34.dp

/** Length of the trailing vine stub past the last entry. */
private val TrailingTailDp: Dp = 40.dp

// endregion

@Composable
fun ActivityTimeline(
    entries: List<ActivityEntry>,
    modifier: Modifier = Modifier,
    header: String = "Activity",
    showsArrow: Boolean = false,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(AppColors.CardSolid)
            .border(0.5.dp, AppColors.BorderDefault, RoundedCornerShape(14.dp)),
    ) {
        // Eyebrow header — matches the YOUR LIBRARY / CONTACTS pattern
        // from the other Home cards.
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(
                    start = AppSpacing.s4,
                    end = AppSpacing.s4,
                    top = AppSpacing.s4,
                    bottom = AppSpacing.s2,
                ),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = header,
                style = AppTypography.Eyebrow,
                color = AppColors.ThemeGreenDeep,
                modifier = Modifier.weight(1f),
            )
            if (showsArrow) {
                Text(
                    text = "\u2192",
                    style = AppTypography.Button,
                    color = AppAccent.primary,
                )
            }
        }

        Spacer(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = AppSpacing.s4)
                .height(0.5.dp)
                .background(AppColors.BorderDefault),
        )

        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(EntryStrideDp * entries.size + TrailingTailDp),
        ) {
            VineDecoration(
                entryCount = entries.size,
                modifier = Modifier.fillMaxWidth(),
            )

            Column {
                entries.forEach { entry ->
                    EntryRow(entry)
                }
            }
        }
    }
}

@Composable
private fun EntryRow(entry: ActivityEntry) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .height(EntryStrideDp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        // Lead column: position the marker at (VineXDp, mid-row) so it
        // lands on the vine line drawn by the Canvas underneath.
        val markerHalf = MarkerSizeDp / 2
        Box(
            modifier = Modifier
                .width(LeadColumnDp)
                .height(EntryStrideDp),
        ) {
            FlowerMarker(
                theme = entry.theme,
                prominence = entry.prominence,
                modifier = Modifier
                    .align(Alignment.TopStart)
                    .offset(
                        x = VineXDp - markerHalf,
                        y = EntryStrideDp / 2 - markerHalf,
                    ),
            )
        }

        Column(
            verticalArrangement = Arrangement.spacedBy(2.dp),
            modifier = Modifier
                .weight(1f)
                .padding(start = 12.dp, end = AppSpacing.s4),
        ) {
            Text(
                text = entry.date,
                fontSize = 11.sp,
                color = AppColors.TextSecondary,
                style = AppTypography.Tag,
            )
            Text(
                text = entry.title,
                fontSize = 14.sp,
                color = AppColors.TextPrimary,
                style = AppTypography.Body,
            )
            entry.preview?.let { preview ->
                Text(
                    text = preview,
                    fontSize = 12.sp,
                    color = AppColors.TextSecondary,
                    style = AppTypography.Meta,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }
    }
}

// region Vine + garland drawing

@Composable
private fun VineDecoration(entryCount: Int, modifier: Modifier = Modifier) {
    val vineGreenDeep = AppColors.ThemeGreenDeep
    val vineGreen = AppColors.ThemeGreenPrimary
    val midribColor = Color(0x66463C31)
    val coralBerry = AppColors.Coral700
    val brownBerry = AppColors.ThemeDryDeep

    Canvas(modifier = modifier.height(EntryStrideDp * entryCount + TrailingTailDp)) {
        if (entryCount == 0) return@Canvas

        val vineX = VineXDp.toPx()
        val bulge = VineBulgeDp.toPx()
        val stride = EntryStrideDp.toPx()
        val totalHeight = stride * entryCount + TrailingTailDp.toPx()

        fun entryY(index: Int): Float = stride / 2f + index * stride

        // --- vine path ---
        val vinePath = Path().apply {
            moveTo(vineX, 0f)
            lineTo(vineX, entryY(0))
            for (index in 1 until entryCount) {
                val prevY = entryY(index - 1)
                val nextY = entryY(index)
                val bulgeRight = (index - 1) % 2 == 0
                val controlX = if (bulgeRight) vineX + bulge else vineX - bulge
                val controlY = (prevY + nextY) / 2f
                quadraticTo(controlX, controlY, vineX, nextY)
            }
            lineTo(vineX, totalHeight)
        }
        drawPath(
            path = vinePath,
            color = vineGreenDeep,
            style = Stroke(width = VineStrokeDp.toPx(), cap = StrokeCap.Round),
        )

        // --- garland: leaf + berry trio in each gap between entries ---
        for (gap in 0 until (entryCount - 1)) {
            val bulgeRight = gap % 2 == 0
            val apexY = (entryY(gap) + entryY(gap + 1)) / 2f
            val sideX = if (bulgeRight) vineX + bulge / 2f else vineX - bulge / 2f

            drawLeaf(
                base = Offset(x = sideX, y = apexY - 14f),
                rotateDegrees = if (bulgeRight) 40f else -40f,
                fill = vineGreen,
                stroke = vineGreenDeep,
                midrib = midribColor,
            )

            drawBerryTrio(
                center = Offset(x = sideX, y = apexY + 30f),
                color = if (bulgeRight) coralBerry else brownBerry,
                mirrored = !bulgeRight,
            )
        }
    }
}

private fun DrawScope.drawLeaf(
    base: Offset,
    rotateDegrees: Float,
    fill: Color,
    stroke: Color,
    midrib: Color,
) {
    // Leaf path drawn at origin, base at (0,0), tip at (0,-48). Fill +
    // stroke + midrib. Wrapped in translate(...) + rotate(...) so the
    // path units stay in the same coordinate space across calls.
    translate(left = base.x, top = base.y) {
        rotate(degrees = rotateDegrees, pivot = Offset.Zero) {
            val leaf = Path().apply {
                moveTo(0f, 0f)
                quadraticTo(-13f, -26f, 0f, -48f)
                quadraticTo(13f, -26f, 0f, 0f)
                close()
            }
            drawPath(path = leaf, color = fill)
            drawPath(path = leaf, color = stroke, style = Stroke(width = 1f))

            val midribPath = Path().apply {
                moveTo(0f, -6f)
                lineTo(0f, -42f)
            }
            drawPath(path = midribPath, color = midrib, style = Stroke(width = 0.8f))
        }
    }
}

private fun DrawScope.drawBerryTrio(center: Offset, color: Color, mirrored: Boolean) {
    val signX = if (mirrored) -1f else 1f
    val offsets = listOf(
        Triple(0f,            0f,    7.2f),
        Triple(10.8f * signX, 6.8f,  6.8f),
        Triple(-5.2f * signX, 10.0f, 6.2f),
    )
    for ((dx, dy, radius) in offsets) {
        drawCircle(
            color = color,
            radius = radius,
            center = Offset(center.x + dx, center.y + dy),
        )
    }
}

// endregion

// region Flower marker

@Composable
private fun FlowerMarker(
    theme: AccentPaletteId,
    prominence: ActivityProminence,
    modifier: Modifier = Modifier,
) {
    val palette = AccentPalettes.forId(theme)
    val isFeatured = prominence == ActivityProminence.Featured
    val petalRx = if (isFeatured) 10.8f else 9.6f
    val petalRy = if (isFeatured) 15.6f else 14.0f
    val centerR = if (isFeatured) 6.8f else 5.6f
    val petalCenterY = if (isFeatured) -17.0f else -15.4f
    val centerColor = if (theme == AccentPaletteId.Yellow) AppColors.Coral700 else AppColors.ThemeYellowPrimary

    Canvas(modifier = modifier.size(MarkerSizeDp)) {
        val cx = size.width / 2f
        val cy = size.height / 2f
        translate(left = cx, top = cy) {
            for (i in 0 until 5) {
                rotate(degrees = i * 72f, pivot = Offset.Zero) {
                    drawOval(
                        color = palette.primary,
                        topLeft = Offset(-petalRx, petalCenterY - petalRy),
                        size = androidx.compose.ui.geometry.Size(petalRx * 2f, petalRy * 2f),
                    )
                    drawOval(
                        color = palette.deep,
                        topLeft = Offset(-petalRx, petalCenterY - petalRy),
                        size = androidx.compose.ui.geometry.Size(petalRx * 2f, petalRy * 2f),
                        style = Stroke(width = 1f),
                    )
                }
            }
            drawCircle(color = centerColor, radius = centerR, center = Offset.Zero)
        }
    }
}

// endregion

File: ios/Releaf/DesignSystem/Components/ActivityTimeline.swift
/*
 * ActivityTimeline.swift
 *
 * Bramble-garland activity timeline. A serpentine vine runs down the
 * left of a cream card. Each entry is a 5-petal flower in one of the
 * four leaf themes; between entries the vine carries a stout leaf and
 * a berry trio sitting on the vine line.
 *
 * Source spec: design-system/timeline-vine-bramble-garland.html.
 *
 * Layout is fixed-height: each entry occupies `entryStride` (90pt) so
 * the vine bezier can be plotted from known y positions. If you need a
 * dynamic-height variant, the cleanest path is to measure rows with a
 * preference key and compute the path on the fly — kept out of v1 to
 * keep the visual diff against the static reference at zero.
 */

import SwiftUI

// MARK: - Public model

public enum ActivityProminence: Sendable {
    /// Featured entry — bigger flower (today, captured highlights).
    case featured
    /// Routine entry — smaller flower.
    case routine
}

public struct ActivityEntry: Identifiable, Sendable {
    public let id: UUID
    public let date: String
    public let title: String
    public let preview: String?
    public let theme: AccentPaletteID
    public let prominence: ActivityProminence

    public init(
        id: UUID = UUID(),
        date: String,
        title: String,
        preview: String? = nil,
        theme: AccentPaletteID,
        prominence: ActivityProminence = .routine
    ) {
        self.id = id
        self.date = date
        self.title = title
        self.preview = preview
        self.theme = theme
        self.prominence = prominence
    }
}

// MARK: - Component

public struct ActivityTimeline: View {
    public let entries: [ActivityEntry]
    public let header: String
    public let showsArrow: Bool

    /// Width of the leading column reserved for the vine + flower marker.
    private let leadColumn: CGFloat = 64
    /// Where the vine sits inside each row (x within the row's leading column).
    private let vineX: CGFloat = 46
    /// Half-width of each vine bulge (controlX offset from vineX).
    private let bulge: CGFloat = 20
    /// Vertical distance between entry centers.
    private let entryStride: CGFloat = 64
    /// Visible flower marker size — used to offset it onto the vine line.
    private let markerSize: CGFloat = 34
    /// Length of the trailing vine stub past the last entry.
    private let trailingTail: CGFloat = 40

    public init(
        entries: [ActivityEntry],
        header: String = "Activity",
        showsArrow: Bool = false
    ) {
        self.entries = entries
        self.header = header
        self.showsArrow = showsArrow
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Eyebrow header — matches the YOUR LIBRARY / CONTACTS
            // pattern from the other Home cards.
            HStack(alignment: .firstTextBaseline) {
                Text(header)
                    .font(AppText.eyebrow)
                    .tracking(AppLetterSpacing.eyebrow)
                    .foregroundStyle(AppColors.themeGreenDeep)
                Spacer()
                if showsArrow {
                    Text("\u{2192}")
                        .font(AppText.button)
                        .foregroundStyle(AppColors.coral)
                }
            }
            .padding(.horizontal, AppSpacing.s4)
            .padding(.top, AppSpacing.s4)
            .padding(.bottom, AppSpacing.s2)

            Rectangle()
                .fill(AppColors.borderDefault)
                .frame(height: 0.5)
                .padding(.horizontal, AppSpacing.s4)

            ZStack(alignment: .topLeading) {
                Canvas { context, _ in
                    drawVine(in: context)
                    drawGarland(in: context)
                }
                .frame(height: timelineHeight)

                VStack(spacing: 0) {
                    ForEach(entries) { entry in
                        entryRow(entry)
                    }
                }
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(AppColors.cardSolid)
        .overlay(
            RoundedRectangle(cornerRadius: 14, style: .continuous)
                .stroke(AppColors.borderDefault, lineWidth: 0.5)
        )
        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
    }

    // MARK: - Layout helpers

    /// Total height of the canvas drawing area: room for every row plus a
    /// short tail so the vine doesn't end abruptly at the last flower.
    private var timelineHeight: CGFloat {
        guard !entries.isEmpty else { return 0 }
        return CGFloat(entries.count) * entryStride + trailingTail
    }

    /// The y at which the vine should reach each entry's flower center.
    private func vineY(for index: Int) -> CGFloat {
        entryStride / 2 + CGFloat(index) * entryStride
    }

    @ViewBuilder
    private func entryRow(_ entry: ActivityEntry) -> some View {
        HStack(alignment: .top, spacing: 0) {
            // Leading column: explicitly position the marker at (vineX, mid)
            // so it lands on the vine line drawn by the Canvas underneath.
            ZStack(alignment: .topLeading) {
                Color.clear
                FlowerMarker(theme: entry.theme, prominence: entry.prominence)
                    .frame(width: markerSize, height: markerSize)
                    .offset(
                        x: vineX - markerSize / 2,
                        y: entryStride / 2 - markerSize / 2
                    )
            }
            .frame(width: leadColumn, height: entryStride)

            VStack(alignment: .leading, spacing: 2) {
                Text(entry.date)
                    .font(.system(size: 11))
                    .foregroundStyle(AppColors.textSecondary)
                Text(entry.title)
                    .font(.system(size: 14))
                    .fontWeight(.medium)
                    .foregroundStyle(AppColors.textPrimary)
                if let preview = entry.preview {
                    Text(preview)
                        .font(.system(size: 12))
                        .foregroundStyle(AppColors.textSecondary)
                        .lineLimit(1)
                        .truncationMode(.tail)
                }
            }
            .padding(.leading, 12)
            .padding(.top, (entryStride - 50) / 2)
            .padding(.trailing, AppSpacing.s4)

            Spacer(minLength: 0)
        }
        .frame(height: entryStride)
    }

    // MARK: - Drawing

    private func drawVine(in context: GraphicsContext) {
        guard !entries.isEmpty else { return }
        var path = Path()
        path.move(to: CGPoint(x: vineX, y: 0))
        for index in entries.indices {
            let y = vineY(for: index)
            if index == 0 {
                path.addLine(to: CGPoint(x: vineX, y: y))
            } else {
                let prevY = vineY(for: index - 1)
                let bulgeRight = (index - 1) % 2 == 0
                let controlX = bulgeRight ? vineX + bulge : vineX - bulge
                let controlY = (prevY + y) / 2
                path.addQuadCurve(
                    to: CGPoint(x: vineX, y: y),
                    control: CGPoint(x: controlX, y: controlY)
                )
            }
        }
        path.addLine(to: CGPoint(x: vineX, y: timelineHeight))

        context.stroke(
            path,
            with: .color(AppColors.themeGreenDeep),
            style: StrokeStyle(lineWidth: 1.8, lineCap: .round)
        )
    }

    private func drawGarland(in context: GraphicsContext) {
        // For each gap (between entry i and entry i+1) place one leaf at the
        // bulge apex and one berry trio further along the vine below the apex.
        for gap in 0..<(entries.count - 1) {
            let bulgeRight = gap % 2 == 0
            let prevY = vineY(for: gap)
            let nextY = vineY(for: gap + 1)
            let apexY = (prevY + nextY) / 2

            // Leaf base sits on the vine apex at vineX +/- bulge/2; tip
            // angled outward 40deg.
            let leafBaseX = bulgeRight ? vineX + bulge / 2 : vineX - bulge / 2
            let leafBase = CGPoint(x: leafBaseX, y: apexY - 14)
            let leafAngle = Angle.degrees(bulgeRight ? 40 : -40)
            drawLeaf(in: context, at: leafBase, angle: leafAngle)

            // Berries sit on the vine 30pt below the apex; vine x there is
            // approximately vineX +/- (bulge/2 - 2).
            let berryX = bulgeRight ? vineX + bulge / 2 : vineX - bulge / 2
            let berryCenter = CGPoint(x: berryX, y: apexY + 30)
            let berryColor: Color = bulgeRight ? AppColors.coral700 : AppColors.themeDryDeep
            drawBerryTrio(in: context, at: berryCenter, color: berryColor, mirrored: !bulgeRight)
        }
    }

    private func drawLeaf(in context: GraphicsContext, at base: CGPoint, angle: Angle) {
        var leaf = Path()
        leaf.move(to: .zero)
        leaf.addQuadCurve(to: CGPoint(x: 0, y: -48), control: CGPoint(x: -13, y: -26))
        leaf.addQuadCurve(to: .zero, control: CGPoint(x: 13, y: -26))
        leaf.closeSubpath()

        var midrib = Path()
        midrib.move(to: CGPoint(x: 0, y: -6))
        midrib.addLine(to: CGPoint(x: 0, y: -42))

        let transform = CGAffineTransform(translationX: base.x, y: base.y)
            .rotated(by: angle.radians)

        context.fill(leaf.applying(transform), with: .color(AppColors.themeGreenPrimary))
        context.stroke(
            leaf.applying(transform),
            with: .color(AppColors.themeGreenDeep),
            lineWidth: 1.0
        )
        context.stroke(
            midrib.applying(transform),
            with: .color(Color(hex: 0x463C31, alpha: 0.4)),
            lineWidth: 0.8
        )
    }

    private func drawBerryTrio(in context: GraphicsContext, at center: CGPoint, color: Color, mirrored: Bool) {
        // Cluster offsets — mirrored on the x-axis for left bulges so the
        // trio leans toward the card interior, not into the padding.
        let signX: CGFloat = mirrored ? -1 : 1
        let offsets: [(CGPoint, CGFloat)] = [
            (CGPoint(x: center.x,                  y: center.y),       7.2),
            (CGPoint(x: center.x + 10.8 * signX,   y: center.y + 6.8), 6.8),
            (CGPoint(x: center.x - 5.2 * signX,    y: center.y + 10.0), 6.2),
        ]
        for (point, radius) in offsets {
            let circle = Path(ellipseIn: CGRect(
                x: point.x - radius,
                y: point.y - radius,
                width: radius * 2,
                height: radius * 2
            ))
            context.fill(circle, with: .color(color))
        }
    }
}

// MARK: - Flower marker

private struct FlowerMarker: View {
    let theme: AccentPaletteID
    let prominence: ActivityProminence

    var body: some View {
        let palette = AccentPalettes.forID(theme)
        let isFeatured = prominence == .featured
        let petalRx: CGFloat = isFeatured ? 10.8 : 9.6
        let petalRy: CGFloat = isFeatured ? 15.6 : 14.0
        let centerR: CGFloat = isFeatured ? 6.8 : 5.6
        let petalCenterY: CGFloat = isFeatured ? -17.0 : -15.4
        let centerColor: Color = theme == .yellow ? AppColors.coral700 : AppColors.themeYellowPrimary

        ZStack {
            ForEach(0..<5, id: \.self) { index in
                Petal(rx: petalRx, ry: petalRy, centerY: petalCenterY)
                    .fill(palette.primary)
                    .overlay(
                        Petal(rx: petalRx, ry: petalRy, centerY: petalCenterY)
                            .stroke(palette.deep, lineWidth: 1.0)
                    )
                    .rotationEffect(.degrees(Double(index) * 72))
            }
            Circle()
                .fill(centerColor)
                .frame(width: centerR * 2, height: centerR * 2)
        }
        .frame(width: 34, height: 34)
    }
}

private struct Petal: Shape {
    let rx: CGFloat
    let ry: CGFloat
    let centerY: CGFloat

    func path(in rect: CGRect) -> Path {
        let cx = rect.midX
        let cy = rect.midY + centerY
        return Path(ellipseIn: CGRect(
            x: cx - rx,
            y: cy - ry,
            width: rx * 2,
            height: ry * 2
        ))
    }
}

// MARK: - Preview

#Preview {
    ZStack {
        AppColors.canvas.ignoresSafeArea()
        ActivityTimeline(entries: [
            ActivityEntry(
                date: "Today",
                title: "Journaled 3 entries",
                preview: "Morning page on procrastination",
                theme: .coral,
                prominence: .featured
            ),
            ActivityEntry(
                date: "Yesterday",
                title: "Walked 2.4 mi",
                preview: "Loop around the lake before dinner",
                theme: .green
            ),
            ActivityEntry(
                date: "Mar 22",
                title: "Captured a quote",
                preview: "On finding meaning in routine",
                theme: .yellow,
                prominence: .featured
            ),
            ActivityEntry(
                date: "Mar 20",
                title: "Photo journal",
                preview: "Spring shoots in the back garden",
                theme: .dry
            ),
            ActivityEntry(
                date: "Mar 18",
                title: "Voice memo",
                preview: "Notes for the new project pitch",
                theme: .green
            ),
        ])
        .padding()
    }
}
